lib/rubocop/cop/minitest/assert_nil.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Minitest
+      # This cop checks uses of `assert_equal(nil, ..)`
+      #
+      # @example
+      #   # bad
+      #   assert_equal(nil, actual)
+      #   assert_equal(nil, actual, 'the message')
+      #
+      #   # good
+      #   assert_nil(actual)
+      #   assert_nil(actual, 'the message')
+      #
+      class AssertNil < Cop
+        MSG_WITHOUT_WARNING = 'Prefer using `assert_nil(%<argument>s)` over ' \
+        '`assert_equal(nil, %<argument>s)`.'
+        MSG_WITH_WARNING = 'Prefer using `assert_nil(%<argument>s, %<msg>s)` over ' \
+        '`assert_equal(nil, %<argument>s, %<msg>s)`.'
```

Comment:
I think there are other approaches, but this `MSG_*` can be combined into one as follows.

```diff
 lib/rubocop/cop/minitest/assert_nil.rb | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/lib/rubocop/cop/minitest/assert_nil.rb b/lib/rubocop/cop/minitest/assert_nil.rb
index e8dcde2..a84e4eb 100644
--- a/lib/rubocop/cop/minitest/assert_nil.rb
+++ b/lib/rubocop/cop/minitest/assert_nil.rb
@@ -15,10 +15,8 @@ module RuboCop
       #   assert_nil(actual, 'the message')
       #
       class AssertNil < Cop
-        MSG_WITHOUT_WARNING = 'Prefer using `assert_nil(%<argument>s)` over ' \
-        '`assert_equal(nil, %<argument>s)`.'
-        MSG_WITH_WARNING = 'Prefer using `assert_nil(%<argument>s, %<msg>s)` over ' \
-        '`assert_equal(nil, %<argument>s, %<msg>s)`.'
+        MSG = 'Prefer using `assert_nil(%<arguments>s)` over ' \
+              '`assert_equal(nil, %<arguments>s)`.'

         def_node_matcher :assert_equal_with_nil, <<~PATTERN
           (send nil? :assert_equal nil (... $_) (str $_)? )
@@ -26,13 +24,15 @@ module RuboCop

         def on_send(node)
           assert_equal_with_nil(node) do |argument, assertion_message|
-            message = if assertion_message.empty?
-                        format(MSG_WITHOUT_WARNING, argument: argument)
-                      else
-                        format(MSG_WITH_WARNING, argument: argument,
-                                                 msg: "'#{assertion_message.first}'")
-                      end
-            add_offense(node, message: message)
+            assertion_message = if assertion_message.first.is_a?(String)
+                                  "'#{assertion_message.first}'"
+                                else
+                                  assertion_message.first
+                                end
+
+            arguments = [argument, assertion_message].compact.join(', ')
+
+            add_offense(node, message: format(MSG, arguments: arguments))
           end
         end
```

Reply:
Thanks so much. Yeah, I did the update and it is more clear now.